woocommerce/woocommerce-android
# TPP: Built in reader connection flow

Closes: #8137


### Description

The PR adds an essential, happy path for TTP. It doesn't touch any UI at this point

Also, this PR changes the flow. Now the first step will be the checker fragment, both for TPP or BT readers. It's needed for cases when a reader is already connected and no need to ask for the reader type

### Testing instructions

Important to test:
* ON release build. Make sure that there is no regression regarding Firebase usage or external readers IPP usage
* ON pre-28 SDK

The flow
* Initiate IPP
* Run both external reader and built-in reader flows

### Images/gif



https://user-images.githubusercontent.com/4923871/212672207-d0d1c9af-18fe-439b-8a68-630190ca248c.mp4
